### ISSUE TITLE
[performance] Symfony Kernel::boot in dev mode on Alpine

### ISSUE THREAD

**Issue by bastnic** [symfony/recipes#712]
This is a follow-up of [this discussion](https://github.com/symfony/recipes/pull/705#discussion_r358711700).

Alpine is the default docker linux base distribution. It's widely used. It's based on musl libc, and so, it's quite different of traditional linux distribution as Debian. One the difference is that glob does not handle `GLOB_BRACE`.

I knew about it from [an old issue on composer](https://github.com/composer/composer/pull/7444) that we already experienced. 

On a big project, we had quite castrophous boot time, and I finally checked on it. 

The first thing is that `Iterator/SortableIterator.php/44-46` cost an enormous amout of time and is called 22k time.

Searching why, It seems that we never enter is the optimized path here https://github.com/symfony/symfony/blob/8ffdef2958d4190994d7540013925b9580e03493/src/Symfony/Component/Config/Resource/GlobResource.php#L102.

I had that in `Kernel::configureContainer`:

```php
const CONFIG_EXTS = '.{php,xml,yaml,yml}';
// ... 
$loader->load($confDir . '/{packages}/*' . self::CONFIG_EXTS, 'glob'); 
$loader->load($confDir . '/{packages}/' . $this->environment . '/**/*' . self::CONFIG_EXTS, 'glob');
$loader->load($confDir . '/{services}' . self::CONFIG_EXTS, 'glob');
```

and in `services.yaml`:

```yaml
Lib\Core\:
    resource: '../src/*'
    exclude: '../src/{Entity,Migrations,Translation,Tests,Kernel.php,PHPStan,Fixtures,Traits}'
```

As I don't have `GLOB_BRACE` and every pattern given contains a brace, it always fallback on the `symfony/finder` implementation which is wayyy slower.

By removing the `/**/*`, and only searching into `yaml` extension, I expect to be in the fast path. But the brackets on `{packages}` and `{services}` prevents it. By removing them it goes into the fast path. And gives a clear boost.

```php
$loader->load($confDir . '/{packages}/*' . self::CONFIG_EXTS, 'glob');

if (is_dir($confDir . '/packages/' . $this->environment)) {
	$loader->load($confDir . '/{packages}/' . $this->environment . '/*' . self::CONFIG_EXTS, 'glob');
}
$loader->load($confDir . '/services.yaml');
```

I didn't see any drwabacks (and I have a very extensive tests suites).

Rest the case with multiple option separated by comma in brackets, that stills cost a lot.

I'm absolutly thinks this is a serious issue, but I'm not sure of what todo with it, hance the absence of PR.

- brackets are they really necessary for packages and services?
- should we purpose a better implementation on that hot path than the `symfony/finder` one. @nicolas-grekas purpose to "we could have some logic that splits a single glob-with-braces into several of them" https://github.com/symfony/recipes/pull/705#discussion_r358748699

**Comment by nicolas-grekas** [comment]
> brackets are they really necessary for packages and services?

Yes, I explained why in https://github.com/symfony/recipes/pull/705#discussion_r358718675
we could remove some of them, but that would make the defaults more rigid - ie favoring perf for DX. We can do better.

> "we could have some logic that splits a single glob-with-braces into several of them"

I confirm this one, it should be that hard.

**Comment by stof** [comment]
I'm transferring this issue to the symfony repo, as that's not specifically about recipes.
